### DISCUSSION TITLE
GetStaticPaths & GetStaticProps: Why is my posts/[id].tsx page running during build? Video and github repo attached

### DISCUSSION THREAD

**Discussion by vladnicula** [vercel/next.js#38193] [Help] [answered]
I'm having a hard time understanding why the page function for a dynamic page is running at build time when the getStaticPaths return an empty paths array. The setup that I have is isolated and really minimal. Check it out below. Any comments and suggestions are welcome. 

Video with the behaviour:
https://www.loom.com/share/5ec3795e075d4c54aef32723e99b007a

GIT Repo with the behaviour:
https://github.com/vladnicula/get-static-props-strange-behaviour

**Comment by icyJoseph** [comment, marked as answer]
Hi,

You have `fallback: true`, which means Next.js is generating a fallback page. More info here, https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-pages

You have to check if `useRouter().isFallback` is `true`, and render something that tells the user that the content is being generated.

### Why?

If a page is not generated, Next.js shows the fallback while it is being prepared. This happens mostly when a user navigates from an external URL to your application, and hits a URL for a page that doesn't exist, but that implements `fallback: true`, then to avoid showing a blank empty void, while the page is prepared, the fallback page is placed, and then when the page is ready, you can show the content.

**Reply by vladnicula** [reply]
Thanks! 

This makes sense. Is the page going to be updated dynamically without having to refresh when the content is ready?

**Reply by icyJoseph** [reply]
Yeah, the value of `useRouter().isFallback` will change when it receives the props (as JSON, you can even see this in the network tab), and then you can do as usual. No refresh of the browser is needed.
